From d403b989621d923287a041dfa19d4a4a1b5fb9e7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 7 Jun 2020 17:36:39 +0200
Subject: [PATCH] Handle indexing on Python 3.8 AST string constants correctly

String constants are represented as ast.Constant objects starting with
Python 3.8.

Fixes #619.
---
 UM/Settings/SettingFunction.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/UM/Settings/SettingFunction.py b/UM/Settings/SettingFunction.py
index 37e7ea52..8172ac7c 100644
--- a/UM/Settings/SettingFunction.py
+++ b/UM/Settings/SettingFunction.py
@@ -245,6 +245,8 @@ class _SettingExpressionVisitor(ast.NodeVisitor):
     def visit_Subscript(self, node: ast.Index):
         if type(node.value) == ast.Str:
             raise IllegalMethodError("Indexing on strings is not allowed")
+        if type(node.value) == ast.Constant and isinstance(node.value.value, str):
+            raise IllegalMethodError("Indexing on strings is not allowed")
         for child_node in ast.iter_child_nodes(node):
             self.visit(child_node)
 
-- 
2.26.2

